mat/band.go

```diff
@@ -136,6 +136,12 @@ func NewBandDense(r, c, kl, ku int, data []float64) *BandDense {
 	}
 }
 
+// NewDiagonal is a convenience function that returns a diagonal matrix represented by a
+// BandDense. The length of data must be min(r, c) otherwise NewDiagonal will panic.
+func NewDiagonal(r, c int, data []float64) *BandDense {
```

Comment:
The sigma in SVD is potentially non-square. Similarly in GSVD. `NewDiagonalBand` and `NewDiagonalSym`? The reason this is here is purely for legibility to draw attention to the fact that the result is a diagonal.